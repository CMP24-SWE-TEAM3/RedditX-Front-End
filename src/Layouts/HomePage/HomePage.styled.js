import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  max-width: 100%;
  box-sizing: border-box;
  flex-direction: row;
  justify-content: center;
  margin: 0 auto;
  @media (min-width: 640px) {
    padding: 20px 24px;
  }

  & .content-posts {
    @media (min-width: 960px) {
      width: 640px;
    }
  }

  & .posts {
    min-height: 1000px;
    width: 100%;
  }

  & aside {
    margin-left: 24px;
    margin-top: 0;

    @media (min-width: 960px) {
      display: block;
    }

    @media (max-width: 970px) {
      display: none;
    }
  }

  & .sidebar {
    display: flex;
    height: 100%;
    flex-direction: column;
    width: inherit;
  }

  & .create-community {
    margin-top: 16px;
    width: 312px;
  }

  & footer {
    flex: 1 1 auto;
    position: relative;
    width: inherit;
  }
`
export const MainContainer = styled.div`
  z-index: 3;

  & .main {
    max-width: 100%;
  }
`