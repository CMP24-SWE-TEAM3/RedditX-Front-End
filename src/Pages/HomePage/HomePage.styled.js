import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  padding-top: 30px;
  max-width: 100%;
  width: 100%;
  box-sizing: border-box;
  flex-direction: row;
  justify-content: center;
  margin: 0 auto;
  overflow-x: hidden;


  //@media (min-width: 640px) {
  //  padding: 20px 24px;
  //  width:95%
  //  
  //}

  & .content-posts {
    width: 100%;
    min-width: 640px;

    @media (max-width: 970px) {
      width:95%;
      //padding-top:48px;

    }
    @media(max-width:640px ){
      width:100%;
      min-width: 100%;
    }
  }

  //& .posts {
  //  min-height: 1000px;
  //  width: 100%;
  //} 

  & aside {
    margin-left: 24px;
    margin-top: 0;

    @media (min-width: 960px) {
      display: block;
    }

    @media (max-width: 970px) {
      display: none;
    }
  }

  .sidebar {
    display: flex;
    height: 100%;
    flex-direction: column;
    width: inherit;
  }
  
  // create community styles
  .create-community {
    margin-top: 16px;
    width: 312px;
  }

  // recent posts styles
  .recent-posts {
    margin-top: 16px;
    width: 312px;
  }
  
  // footer styles
  & footer {
    flex: 1 1 auto;
    position: relative;
    width: inherit;
  }

  & .child-footer {
    position: sticky;
    top: 57px;
  }

  & .scroll-but {
    top: calc(100vh - 8px);
    display: flex;
    margin-top: 45px;
    justify-content: center;
    text-align: center;
    transform: translateY(-100%);
  }
`
export const MainContainer = styled.div`
  @media (max-width: 960px) {
    width: 100%;
  }

  & .main {
    max-width: 100%;
  }
`