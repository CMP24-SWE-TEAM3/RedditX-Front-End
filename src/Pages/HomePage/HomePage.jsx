import React, {Component, useState} from 'react';
import {Container, MainContainer} from "./HomePage.styled";
import Navbar from "Layouts/Navbar/Navbar";
import CreatePost from "Layouts/CreatePost/CreatePost";
import TopCommunities from "Components/CommunityCard/CommunityCard";
import PopularPosts from "Layouts/PopularPosts/PopularPosts";
import CreatePostSideBar from "Layouts/CreatePostSideBar/CreatePostSideBar";
import Footer from "Layouts/Footer/Footer";
import * as styles from "App.styled";
import lightTheme from "Theme/lightTheme";
import defaultTheme from "Theme/defaultTheme";
import darkTheme from "Theme/darkTheme";
import {ThemeProvider} from "styled-components";
import ScrollButton from "Components/ScrollButton/ScrollButton";
import {SideBar} from "Components/DropDownButtonLeft/DropDownLeft";

/**
 * Component that displays a list of layouts such as  posts , navigation , and sidebar.
 *
 * @returns {React.Component} returns an instance of HomePage with a list of layouts created
 */

const HomePage = () => {

    const [theme, setTheme] = useState({...defaultTheme, ...lightTheme});
    return (

        <ThemeProvider theme={theme}>
            <styles.AppContainer>
                <styles.AppHeader>
                    <Navbar/>
                    <MainContainer>
                        {/*<SideBar/>*/}
                        <div className="main"/>
                        <Container>
                            <div className={'content-posts'}>
                                <CreatePost/>
                                <PopularPosts/>
                                {/*<div className={'posts'}>*/}
                                {/*</div>*/}
                            </div>
                            <aside>
                                <div className={'sidebar'}>
                                    <TopCommunities/>

                                    <div className={'create-community'}>
                                        <CreatePostSideBar/>
                                    </div>
                                    <footer>
                                        <div className={'child-footer'}>
                                            <Footer/>
                                        </div>
                                        <div className={'scroll-but'}>
                                            <ScrollButton/>
                                        </div>
                                    </footer>
                                </div>
                            </aside>
                        </Container>
                    </MainContainer>
                </styles.AppHeader>
            </styles.AppContainer>
        </ThemeProvider>
    );
};

export default HomePage;
