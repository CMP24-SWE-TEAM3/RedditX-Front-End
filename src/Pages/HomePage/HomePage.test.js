import {shallow} from 'enzyme';

// import Components
import HomePage from "./HomePage";

describe("Home Page", () => {
    it("this is a test for Home Page", () => {
        expect(shallow(<HomePage/>)).toMatchSnapshot();

    });
});