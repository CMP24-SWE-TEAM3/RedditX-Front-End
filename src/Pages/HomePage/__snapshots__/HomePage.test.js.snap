// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page this is a test for Home Page 1`] = `ShallowWrapper {}`;
